### PR TITLE
Use identifiers as symbols to uniquely identify these rmw implementations

### PR DESCRIPTION
**Pull request by esteve**
Connects to ros2/ros2#18

### CODE REVIEW

**Comment by esteve** [comment]
@dirk-thomas @tfoote @wjwwood ping

File: rmw_connext_xtypes_dynamic_cpp/include/rmw_connext_xtypes_dynamic_cpp/impl/rmw/get_type_support_handle.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+extern "C" {
+  extern const char * rti_connext_dynamic_identifier;
+}
+
+namespace rmw
+{
+template<typename T>
+const rosidl_message_type_support_t * get_type_support_handle()
+{
+  std::cout << rti_connext_dynamic_identifier << std::endl;
```
Comment (dirk-thomas) [inline rmw_connext_xtypes_dynamic_cpp/include/rmw_connext_xtypes_dynamic_cpp/impl/rmw/get_type_support_handle.h]: Remove debug output?

File: rosidl_typesupport_connext_cpp/cmake/rosidl_typesupport_connext_cpp_generate_interfaces.cmake [hunk 2]
```diff
@@ -129,5 +129,6 @@ install(
   DESTINATION "lib"
 )
 
-ament_export_include_directories(include)
 ament_export_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} ${Connext_LIBRARIES})
+
+ament_export_include_directories(include)
```
Comment (dirk-thomas) [inline rosidl_typesupport_connext_cpp/cmake/rosidl_typesupport_connext_cpp_generate_interfaces.cmake]: Why moving this line around? The previous order (include before libraries) seems to make sense.

File: rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt [hunk 3]
```diff
@@ -19,7 +19,7 @@ find_package(rosidl_generator_cpp REQUIRED)
 
 find_package(rosidl_typesupport_introspection_cpp REQUIRED)
 ament_export_dependencies(rosidl_typesupport_introspection_cpp)
-
+ament_export_include_directories(include/${PROJECT_NAME}/impl ${Connext_INCLUDE_DIRS})
```
Comment (dirk-thomas) [inline rmw_connext_xtypes_dynamic_cpp/CMakeLists.txt]: I would suggest moving the include dir export either above or below both dependency exports.

File: rosidl_typesupport_connext_cpp/cmake/rosidl_typesupport_connext_cpp_generate_interfaces.cmake (hunk 2 above)
Comment (esteve) [inline rosidl_typesupport_connext_cpp/cmake/rosidl_typesupport_connext_cpp_generate_interfaces.cmake]: To make it more consistent with `rmw_opensplice`. I also was going to add support for "native" `Duration_t` and `Time_t`, so it needs to skip most of the CMake file if the package name is `builtin_msgs`

Comment (dirk-thomas) [inline rosidl_typesupport_connext_cpp/cmake/rosidl_typesupport_connext_cpp_generate_interfaces.cmake]: Makes sense.

**Comment by tfoote** [comment]
lgtm

**Comment by dirk-thomas** [comment]
lgtm

**Pull request merged**